src/OAuth/ResourceOwner/PassageResourceOwner.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the HWIOAuthBundle package.
+ *
+ * (c) Hardware Info <opensource@hardware.info>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace HWI\Bundle\OAuthBundle\OAuth\ResourceOwner;
+
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+final class PassageResourceOwner extends GenericOAuth2ResourceOwner
+{
+    public const TYPE = 'passage';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected array $paths = [
+        'identifier' => 'sub',
+        'email' => 'email',
+        'phone_number' => 'phone_number',
+        'email_verified' => 'email_verified',
+        'phone_number_verified' => 'phone_number_verified',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function revokeToken($token)
+    {
+        if (!isset($this->options['revoke_token_url'])) {
+            throw new AuthenticationException('OAuth error: "Method unsupported."');
+        }
+
+        $parameters = [
+            'client_id' => $this->options['client_id'],
+            'client_secret' => $this->options['client_secret'],
+            'token' => $token,
+        ];
+
+        $response = $this->httpRequest($this->normalizeUrl($this->options['revoke_token_url']), $parameters, [], 'POST');
+
+        return 200 === $response->getStatusCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $resolver->setDefaults([
+            'authorization_url' => 'https://{sub_domain}.withpassage.com/authorize',
+            'access_token_url' => 'https://{sub_domain}.withpassage.com/token',
+            'revoke_token_url' => 'https://{sub_domain}.withpassage.com/revoke',
+            'infos_url' => 'https://{sub_domain}.withpassage.com/userinfo',
+
+            'use_commas_in_scope' => false,
+            'scope' => 'openid email phone',
```

Comment:
As phone is not mandatory & IMO not so common, maybe put this as an option in documentation instead of adding it as default?

Reply:
In the case of Passage, `email` is not mandatory either, just `openid`.
But you're right, `phone` is not really common.
At least you have to choose in the [Passage backend](https://console.passage.id) if a user is allowed to register with only email, only phone or if the user can choose between both. At least you'll have to set one or both in the configuration of the Passage backend.
https://docs.passage.id/hosted-login/oidc-client-configuration#supported-scopes

At the end the scope will just define which user information will Passage send you.

Reply:
Email is not required by the provider, but it's most likely required by this bundle and user providers as an identifier.

Reply:
Ok, changed it.
Is it ok to just mention to check out the documentation for different scopes, or should I add it to the provider documentation in this repository?
https://github.com/hwi/HWIOAuthBundle/pull/1962/files#diff-b41f069f6f25945f90e04d985da7846922971c5c0bfafbf336c92d57df2b1b3aR8

Reply:
For me it's good enough, but you need to fix tests as there is no phone scope in calls